### ISSUE TITLE
Issues with 512px zoom 8 128/84 tile (Colchester, England)

### ISSUE THREAD

**Issue by Unigoge** [tilezen/vector-datasource#1481]
* **What did you see?**
Ocean coastline is missing on MVT tile
Ocean coastline includes at least one wrong polygon on JSON tile (probably a result of polygons unification)

* **What did you expect to see?**

an Ocean coastline - it is getting rendered properly on TopoJSON tile :)

* **What map location are you having problems with?** 
512px zoom 8 128/84 tile (Colchester, England)

* **Screenshot?** _Props for animated gifs._
good TopoJSON:
![screen shot 2018-03-28 at 3 01 31 pm](https://user-images.githubusercontent.com/17221599/38050466-200f9da0-3299-11e8-878d-4eb6a52e42bb.png)

bad JSON:
![bad JSON](https://user-images.githubusercontent.com/17221599/38049662-d4623b80-3296-11e8-8f84-133caad91256.png)

bad MVT:
![bad MVT](https://user-images.githubusercontent.com/17221599/38049741-fc545b5a-3296-11e8-9a86-447ce6a82c10.png)

**Comment by nvkelso** [comment]
@Unigoge Are you generating the tiles locally or using the Nextzen endpoints or? It looks like the data is correct in that the TopoJSON format shows everything correctly, but GeoJSON and MVT show different levels of wrong. Is this what you're experiencing?

/cc @zerebubuth as this smells like polygon validity funk.

**Comment by Unigoge** [comment]
Yes, all tiles were generated on my server. It was built using 1.4.1 release (and corresponding to it tilequeue and tileserver releases). The database was populated by OSM planet data by the end of February.

**Comment by m-abboud** [comment]
Have the same issue with 1.4.3 and did my import in march. This also happens on a tile to the right of Boston at zoom 9 155/189

**Comment by m-abboud** [comment]
Figured it out (or how to fix it at least).  Found out tilequeue defaults to 256 size for non zipped meta tiles. So if you're just caching mvt's by themselves it uses the 256 size which I assume you're also doing since we had this same exact problem. I'm not really sure why this would cause the ocean to be missing entirely from some tiles though but it's definitely the cause of it.

So to fix it without having to use the zipped meta tiles and tapacatal you can change the default tile size in tiles.py by changing the metatile default zoom from 0 to 1 in two functions at the bottom, metatile_zoom_from_str and metatile_zoom_from_size or see this commit: https://github.com/apollomapping/tilequeue/commit/b2d1ca2d737f26b5bb73e8e07dc0b6c173c708de

Since everyone uses 512 kinda seems like the default should be 512 for the single tiles without having to do that yourself though

**Comment by zerebubuth** [comment]
> I'm not really sure why this would cause the ocean to be missing entirely from some tiles though but it's definitely the cause of it.

The 512px and 256px tiles have different extents, which mean different coordinate transformations for MVT and GeoJSON formats. Both of these are truncating the coordinates in different ways, which can cause polygons to become invalid. There is [a lot of code in our `mapbox-vector-tile` library](https://github.com/tilezen/mapbox-vector-tile/blob/46629e478935954fc2dcbd8d17d005546b8d31a0/mapbox_vector_tile/polygon.py) which tries to make the polygons valid and sensible, but there are [many](https://github.com/tilezen/mapbox-vector-tile/issues/103) [known](https://github.com/tilezen/mapbox-vector-tile/issues/100) [issues](https://github.com/tilezen/mapbox-vector-tile/issues/95) [with](https://github.com/tilezen/mapbox-vector-tile/issues/87) [it](https://github.com/tilezen/mapbox-vector-tile/issues/42).

Using a different tile size (i.e: 512px vs 256px) could help, but it's likely that you'll find similar errors - just in different places. We did a little [proof-of-concept](https://github.com/tilezen/coanacatl) to try and fix it :crossed_fingers:, but haven't got back around to the due diligence necessary before we can use it in `tilequeue`.

> kinda seems like the default should be 512 for the single tiles without having to do that yourself though

If you want to change the tile size without needing code changes, you can use URLs like `http://your.tile.server/mapzen/vector/v1/512/all/z/x/y.mvt` (note the `/512/` between `/v1` and `/all`, which is the `{tilesize}` parameter in [the docs](https://github.com/tilezen/vector-datasource/blob/master/docs/index.md#requesting-tiles)).

As for why it's the default... it's just historical reasons. When we launched the Tilezen service, the tile size was fixed at 256px. When we added the ability to request different tile sizes, we needed to be backwards-compatible with the previous URL structure so that we didn't create problems for existing users.

**Comment by m-abboud** [comment]
Gotcha

But changing url to http://your.tile.server/mapzen/vector/v1/512/all/z/x/y.mvt won't fix tilequeue defaulting to 512 for the cache created by tilequeue that gets hit first before it hits tileserver

**Comment by m-abboud** [comment]
Or more clear, theres no option to change the tile size when you're storing plain mvts vs the zipped 512+256 ones so a code change is necessary for those too time deficiant to setup tapacatl as well.

 And which is why at the top where he says he's using 512 ones and probabley going to the /512/ url he's actually using 256 ones.

**Comment by nvkelso** [comment]
Closing for lack of activity.
